src/ZI2C.cpp

```diff
@@ -46,25 +86,53 @@ ZI2C::ZI2C(ZPin &sda, ZPin &scl) : codal::I2C(sda,scl), sda(sda), scl(scl)
     else
         target_panic(DEVICE_HARDWARE_CONFIGURATION_ERROR);
 
-    DMESG("SDA idx %d, fn: %d", sercomIdx, sda_fun);
-    DMESG("SCL idx %d, fn: %d", sercomIdx, scl_fun);
+    gpio_set_pin_direction(sda_pin->number, GPIO_DIRECTION_OUT);
+    gpio_set_pin_direction(scl_pin->number, GPIO_DIRECTION_OUT);
+    gpio_set_pin_level(sda_pin->number,true);
+    gpio_set_pin_level(scl_pin->number,true);
+
+    gpio_set_pin_function(sda_pin->number, GPIO_PIN_FUNCTION_OFF);
+    gpio_set_pin_function(scl_pin->number, GPIO_PIN_FUNCTION_OFF);
+    gpio_set_pin_direction(sda_pin->number, GPIO_DIRECTION_IN);
+    gpio_set_pin_direction(scl_pin->number, GPIO_DIRECTION_IN);
+
+    gpio_set_pin_pull_mode(sda_pin->number, GPIO_PULL_DOWN);
```

Comment:
there seem to be external 10k pullups on sda/scl, so the 40k internal pullup will probably leave the thing at logic 1. So it would seem there should be no difference between PULL_OFF and PULL_DOWN.

Reply:
@mmoskal I have a hack that works, I'm reluctant to change it despite your advice :smile: